### ISSUE TITLE
Peacock Muting Not Working

### ISSUE THREAD

**Issue by richardfrost** [FrostCo/AdvancedProfanityFilter#452]
#### :bug: Description
Muting is no longer working on Peacock.

#### :pencil: Additional context
Reported by @WolfD1g1tal

**Comment by richardfrost** [comment]
It looks like Peacock has updated their site captions. I think I have worked out the updates and will get it included in the next version of APF, but here is the new config if you'd like to test it out.

```json
{
  "www.peacocktv.com": [
    {
      "displaySelector": "div.video-player__subtitles",
      "mode": "elementChild",
      "parentSelector": "div.video-player__subtitles > span.video-player__subtitles__cue span.video-player__subtitles__line",
      "simpleUnmute": true,
      "tagName": "#text"
    },
    {
      "className": "video-player__subtitles",
      "displaySelector": "div.video-player__subtitles",
      "mode": "element",
      "subtitleSelector": "span.video-player__subtitles__cue > span.video-player__subtitles__line",
      "tagName": "DIV"
    }
  ]
}
```

To use this config, paste it into the Custom Sites area (Options page > Audio tab) and click the blue "SAVE" button underneath it. Then, make sure to refresh the tab for the video site (if you have it open still) and it will load this new configuration.

For best results, make sure to remove any custom config used for testing like this once its been included in the official update.

**Comment by WolfD1g1tal** [comment]
You sir are a steely eyed missile  man. She went green soon as I reloaded and set cc on. I will holler back in a day or two after I’ve watched some different things on different sections of the site.
I am so grateful you got back so quick. Thank You!

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by richardfrost** [comment]
@WolfD1g1tal lol, thanks! I'm glad I was able to hopefully fix it soon. In the meantime, I've updated the [Supported Sites Status Page](https://apf.frostco.dev/status) to reflect that its not currently working. Let me know how your experience goes and if you give it the go-ahead I'll include it in a new update soon. Thanks again for reporting the issue!

**Comment by ahkelly** [comment]
Hi Richard, I am having the muting issue again on Peacock - sorry. The subtitle update is working just not the mute.

**Comment by richardfrost** [comment]
@ahkelly Thanks for reporting the issue. I'll take a look and see if I can get it working again soon.

**Comment by richardfrost** [comment]
Sorry it has taken me so long to get back to you @ahkelly, could you give this config a try:

```json
{
  "www.peacocktv.com": [
    {
      "displaySelector": "div.video-player__subtitles",
      "mode": "watcher",
      "subtitleSelector": "div.video-player__subtitles > span.video-player__subtitles__cue span.video-player__subtitles__line"
    }
  ]
}
```

To use this config, paste it into the Custom Sites area (Options page > Audio tab) and click the blue "SAVE" button underneath it. Then, make sure to refresh the tab for the video site (if you have it open still) and it will load this new configuration.

I tested it on a few videos with success, but would love a second opinion before releasing it in an update.

**Comment by ahkelly** [comment]
Not sure if you made some more updates since I posted this but I finally checked it again today and the muting appears to be working again now without adding the custom site you note above.

**Comment by ahkelly** [comment]
Scratch that, as I continued to watch a program it was sporadically not working. I will add this new custom definition and let you know if that helps so far it has in less than a minute test.

**Comment by richardfrost** [comment]
@ahkelly I went ahead and included the new config in version [3.2.3](https://github.com/richardfrost/AdvancedProfanityFilter/releases/tag/3.2.3). Let me know how it goes though if you get the chance.

**Comment by richardfrost** [comment]
This has been included in version [3.4.0](https://github.com/richardfrost/AdvancedProfanityFilter/releases/tag/3.4.0).
